File: library/src/commonMain/kotlin/com/doordeck/sdk/api/responses/AccountResponses.kt
package com.doordeck.sdk.api.responses

import com.doordeck.sdk.api.model.TwoFactorMethod
import kotlinx.serialization.Serializable
import kotlin.js.JsExport

@JsExport
@Serializable
class TokenResponse(
    val authToken: String,
    val refreshToken: String
)

@JsExport
@Serializable
class UserDetailsResponse(
    val email: String,
    val displayName: String? = null,
    val emailVerified: Boolean,
    val publicKey: String,
)

@JsExport
@Serializable
class RegisterEphemeralKeyResponse(
    val certificateChain: Array<String>,
    val userId: String
)

@JsExport
@Serializable
class RegisterEphemeralKeyWithSecondaryAuthenticationResponse(
    val method: TwoFactorMethod
)

File: doordeck-sdk/src/mingwMain/resources/csharp/Model/Responses/AccountResponses.cs
using System.Security.Cryptography.X509Certificates;

namespace Doordeck.Headless.Sdk.Model.Responses;

public class TokenResponse
{
    public required string AuthToken { get; set; }
    public required string RefreshToken { get; set; }
}

public class UserDetailsResponse
{
    public required string Email { get; set; }
    public string? DisplayName { get; set; }
    public required bool EmailVerified { get; set; }
    public required string PublicKey { get; set; }
}

public class RegisterEphemeralKeyResponse
{
    public required List<X509Certificate> CertificateChain { get; set; }
    public required Guid UserId { get; set; }
}

public class RegisterEphemeralKeyWithSecondaryAuthenticationResponse
{
    public required TwoFactorMethod Method { get; set; }
}